dlt/normalize/items_normalizers.py

```diff
@@ -137,7 +139,6 @@ def _adjust_outer_select_with_dlt_columns(
 
         schema_update: TSchemaUpdate = {}
         schema = self.schema
-        dialect = self.config.destination_capabilities.sqlglot_dialect
```

Comment:
for now I just use the dialect from the model file. We can also make sure that the injected capabilities take the configuration into account with a small change in pipeline, but imho we should probably raise here if model dialect and destination dialect are not the same..